runtime/src/threads/pthreads/threads-pthreads.c

```diff
@@ -81,6 +83,102 @@ static void            (*saved_threadEndFn)(void);
 static void*           initial_pthread_func(void*);
 static void*           pthread_func(void*);
 
+/* This conditional is necessary because on some operating systems
+ * (like OSX) the cstdlib malloc doesn't use nmap for the allocation
+ * and can't be mprotect-ed. In that case, we use the default pthread
+ * stack
+ */
+#ifndef CHPL_USING_CSTDLIB_MALLOC
```

Comment:
Similar to the comment for tasks-fifo.c, I'd prefer we check at execution time for a registered heap by calling chpl_comm_desired _shared_heap(), rather than doing this compile-time check.